overlord/snapstate/progress.go

```diff
@@ -75,13 +88,8 @@ func (t *taskProgressAdapter) Finished() {
 
 // Write sets the current write progress
 func (t *taskProgressAdapter) Write(p []byte) (n int, err error) {
-	if t.unlocked {
-		t.task.State().Lock()
-		defer t.task.State().Unlock()
-	}
-
 	t.current += float64(len(p))
```

Comment:
t.current is assigned twice, in t.Set() and here.

Reply:
yes, that is a bit strange. Wouldn't 

t.Set(t.current + float64(len(p))

work?